install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -362,7 +416,45 @@ spec:
         - name: cni-path
           mountPath: /hostbin
         securityContext:
+          {{- if .Values.securityContext.privileged }}
           privileged: true
+          {{- else }}
+          seLinuxOptions:
+            level: 's0'
+            # Running with spc_t since we have removed the privileged mode.
+            # Users can change it to a different type as long as they have the
+            # type available on the system.
+            type: 'spc_t'
+          capabilities:
+            add:
+              # Only used for 'mount' cgroup
+              - SYS_ADMIN
+              # Used for nsenter
+              - SYS_PTRACE
+          {{- end}}
+      {{- end }}
+      {{- if not .Values.securityContext.privileged }}
+      # Mount the bpf fs if it is not mounted. We will perform this task
+      # from a privileged container because the mount propagation bidirectional
+      # only works from privileged containers.
+      - name: mount-bpf-fs
+        image: {{ include "cilium.image" .Values.image | quote }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args:
+        - 'mount | grep "/sys/fs/bpf type bpf" || mount -t bpf bpf /sys/fs/bpf'
+        command:
+        - /bin/bash
+        - -c
+        - --
+        securityContext:
+          privileged: true
+      {{- /* CRI-O already mounts the BPF filesystem */ -}}
+      {{- if not (eq .Values.containerRuntime.integration "crio") }}
```

Comment:
This part seems like it should be above the whole init container that's defined here..?

Reply:
Good point, however I don't think it hurts running this init container even if cri-o already mounts the BPF filesystem. Let's assume worst case where cri-o mounts the directory inside the container but the bpf fs is not mounted.